pkg/util/pki/subject.go

```diff
@@ -79,8 +79,19 @@ func UnmarshalSubjectStringToRDNSequence(subject string) (pkix.RDNSequence, erro
 
 		atvs := make([]pkix.AttributeTypeAndValue, 0, len(ldapRelativeDN.Attributes))
 		for _, ldapATV := range ldapRelativeDN.Attributes {
+			oid, ok := attributeTypeNames[ldapATV.Type]
+			if !ok {
+				// If the attribute type is not known, we try to parse it as an OID.
+				// If it is not an OID, we set Type=nil
+
+				oid, err = ParseObjectIdentifier(ldapATV.Type)
+				if err != nil {
+					oid = nil
+				}
```

Comment:
What's the significance of `nil` vs `[]int{}`?
Isn't it going to be `nil` anyway (assuming `ParseObjectIdentifier` returns `nil, err`)?

Reply:
Indeed, I think `ParseObjectIdentifier` returns `nil, err`.
I set `oid = nil` here to accentuate that we actually rely on this `oid` value being `nil`.
In https://github.com/cert-manager/cert-manager/blob/1035bb44ca8e9a1cd49c9de79dd5b28af1dcc972/internal/apis/certmanager/validation/certificate.go#L95-L102 we actually rely on this `Type == nil` setup.